### PR TITLE
Add tox and flake8 support

### PR DESCRIPTION
**Pull request by pabelanger**
This is the first of many commits to help make the code more PEP8
compliant.  This patch just enables the tools, but no actuall code
changes yet.

Signed-off-by: Paul Belanger paul.belanger@polybeacon.com

### CODE REVIEW

File: saltcloud/deploy/bootstrap-salt.sh [hunk 1]
```diff
@@ -26,9 +26,7 @@ ScriptName="bootstrap-salt.sh"
 #   * BS_PIP_ALLOWED:     If 1 enable pip based installations(if needed)
 #   * BS_ECHO_DEBUG:      If 1 enable debug echo which can also be set by -D
 #   * BS_SALT_ETC_DIR:    Defaults to /etc/salt
-#   * BS_KEEP_TEMP_FILES: If 1, don't move temporary files, instead copy them
```
Comment (pabelanger) [inline saltcloud/deploy/bootstrap-salt.sh:29]: I'm not sure how this file is changing, I did not edit it.  Is it some how dynamically generated?

Comment (techhat) [inline saltcloud/deploy/bootstrap-salt.sh:29]: Do you mean bootstrap-salt.sh? Locally, this file can be updated with salt-cloud -u. In the repo, it is updated when a new release of salt-cloud is cut. Also, periodically, @s0undt3ch  will update it to match the latest stable release of salt-bootstrap.

Comment (techhat) [inline saltcloud/deploy/bootstrap-salt.sh:29]: You'll want to pull bootstrap-salt.sh out of this PR.

**Comment by pabelanger** [comment]
That should be better, removed the modified bootstrap file.

**Comment by techhat** [comment]
@pabelanger, let's hold off on any more PEP8 fixes for the moment... @s0undt3ch and I are about to make some massive changes to the entire codebase, and I worry that your changes would get lost in the fray and be for naught.

**Pull request merged**